src/Microsoft.VisualStudio.Editors/PropPages/ReferencePropPage.vb

```diff
@@ -792,10 +793,11 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                 Dim solution = visualStudioWorkspace.CurrentSolution
                 Dim CurrentImportNames As New List(Of String)
 
-                For Each projectId In solution.ProjectIds
-                    ' We need to find the project that matches by IVsHierarchy
-                    If visualStudioWorkspace.GetHierarchy(projectId) Is ProjectHierarchy Then
-                        Dim compilationTask = solution.GetProject(projectId).GetCompilationAsync(cancellationTokenSource.Token)
+                For Each project In solution.Projects
```

Comment:
> In fact, I'm convinced that this is broken when running it against legacy project system, attempting to change Imports leads to some very strange behavior.

Legacy project system will be unaffected with this change. This code is merely changing how to identify the Roslyn project corresponding to the project whose Reference page is open. I dont see a reason why this change should affect how changes to imports will be affected.